commands/app/__tests__/migrate.test.ts

```diff
@@ -0,0 +1,135 @@
+const mockedLogger = logger as jest.Mocked<typeof logger>;
+
+describe('commands/app/migrate', () => {
+  const mockAccountId = 123;
+  const mockAccountConfig = {
+    name: 'Test Account',
+    env: 'prod',
+  };
+  let exitSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    exitSpy = jest.spyOn(process, 'exit').mockImplementation();
+    mockedGetAccountConfig.mockReturnValue(mockAccountConfig);
+  });
+
+  describe('handler', () => {
+    it('should exit with error when no account config is found', async () => {
+      mockedGetAccountConfig.mockReturnValue(null);
+
+      await handler({
+        derivedAccountId: mockAccountId,
+      } as ArgumentsCamelCase<MigrateAppOptions>);
+
+      expect(mockedLogger.error).toHaveBeenCalled();
+      expect(exitSpy).toHaveBeenCalledWith(EXIT_CODES.ERROR);
+      exitSpy.mockRestore();
+    });
+
+    it('should call migrateApp2025_2 for platform version 2025.2', async () => {
+      await handler({
+        derivedAccountId: mockAccountId,
+        platformVersion: PLATFORM_VERSIONS.v2025_2,
+      } as ArgumentsCamelCase<MigrateAppOptions>);
+
+      expect(mockedMigrateApp2025_2).toHaveBeenCalledWith(
+        mockAccountId,
+        expect.any(Object)
+      );
+      expect(mockedMigrateApp2023_2).not.toHaveBeenCalled();
+    });
+
+    it('should call migrateApp2023_2 for platform version 2023.2', async () => {
+      await handler({
+        derivedAccountId: mockAccountId,
+        platformVersion: PLATFORM_VERSIONS.v2023_2,
+      } as ArgumentsCamelCase<MigrateAppOptions>);
+
+      expect(mockedMigrateApp2023_2).toHaveBeenCalledWith(
+        mockAccountId,
+        expect.any(Object),
+        mockAccountConfig
+      );
+      expect(mockedMigrateApp2025_2).not.toHaveBeenCalled();
+    });
+
+    it('should handle errors during migration', async () => {
+      const mockError = new Error('Migration failed');
+      mockedMigrateApp2023_2.mockRejectedValue(mockError);
+      const exitSpy = jest.spyOn(process, 'exit').mockImplementation();
+
+      await handler({
+        derivedAccountId: mockAccountId,
+        platformVersion: PLATFORM_VERSIONS.v2023_2,
+      } as ArgumentsCamelCase<MigrateAppOptions>);
+
+      expect(mockedLogger.error).toHaveBeenCalled();
+      expect(exitSpy).toHaveBeenCalledWith(EXIT_CODES.ERROR);
+      exitSpy.mockRestore();
+    });
+  });
+
+  describe('builder', () => {
+    let mockYargs: Argv;
+
+    beforeEach(() => {
+      mockYargs = {
+        options: jest.fn().mockReturnThis(),
+        option: jest.fn().mockReturnThis(),
+        example: jest.fn().mockReturnThis(),
+        conflicts: jest.fn().mockReturnThis(),
+        argv: { _: ['app', 'migrate'] },
+      } as unknown as Argv;
+    });
+
+    it('should add required options', async () => {
+      await builder(mockYargs);
+
+      expect(mockYargs.options).toHaveBeenCalledWith(
+        expect.objectContaining({
+          name: expect.any(Object),
```

Comment:
Does `expect.any(Object)` work with all of these options? I would expect `name` and `dest` to be strings, etc.

Reply:
This is checking the options calls and not the type of the option.  I will expand it because it's a pretty generic test.